anitya/__init__.py

```diff
@@ -1,172 +1,4 @@
-                versions=project.versions,
-                agent='anitya',
-                odd_change=odd_change,
-            ),
-        )
-
-    session.add(project)
-    session.commit()
-
-
-def _construct_substitutions(msg):
-    """ Convert a fedmsg message into a dict of substitutions. """
-    subs = {}
-    for key1 in msg:
-        if isinstance(msg[key1], dict):
-            subs.update(dict([
-                ('.'.join([key1, key2]), val2)
-                for key2, val2 in _construct_substitutions(msg[key1]).items()
-            ]))
-
-        subs[key1] = msg[key1]
-
-    return subs
-
-
-def log(session, project=None, distro=None, topic=None, message=None):
-    """ Take a partial fedmsg topic and message.
-
-    Publish the message and log it in the db.
-    """
-
-    # To avoid a circular import.
-    import anitya.lib.model as model
-
-    # A big lookup of fedmsg topics to model.Log template strings.
-    templates = {
-        'distro.add': '%(agent)s added the distro named: %(distro)s',
-        'distro.edit': '%(agent)s edited distro name from: %(old)s to: '
-                       '%(new)s',
-        'distro.remove': '%(agent)s deleted the distro named: %(distro)s',
-        'project.add': '%(agent)s added project: %(project)s',
-        'project.add.tried': '%(agent)s tried to add an already existing '
-                             'project: %(project)s',
-        'project.edit': '%(agent)s edited the project: %(project)s fields: '
-                        '%(changes)s',
-        'project.flag': '%(agent)s flagged the project: %(project)s with '
-                        'reason: %(reason)s',
-        'project.flag.set': '%(agent)s set flag %(flag)s to %(state)s',
-        'project.remove': '%(agent)s removed the project: %(project)s',
-        'project.map.new': '%(agent)s mapped the name of %(project)s in '
-                           '%(distro)s as %(new)s',
-        'project.map.update': '%(agent)s update the name of %(project)s in '
-                              '%(distro)s from: %(prev)s to: %(new)s',
-        'project.map.remove': '%(agent)s removed the mapping of %(project)s '
-                              'in %(distro)s',
-        'project.version.remove': '%(agent)s removed the version %(version)s '
-                                  'of %(project)s ',
-        'project.version.update': 'new version: %(upstream_version)s found'
-                                  ' for project %(project.name)s '
-                                  '(project id: %(project.id)s).',
-    }
-    substitutions = _construct_substitutions(message)
-    final_msg = templates[topic] % substitutions
-
-    fedmsg_publish(topic=topic, msg=dict(
-        project=project,
-        distro=distro,
-        message=message,
-    ))
-
-    model.Log.insert(
-        session,
-        user=message['agent'],
-        project=project,
-        distro=distro,
-        description=final_msg)
-
-    return final_msg
```

Comment:
Nope, this breaks APIs, but there's not a good way around it. We're still versioned 0.y.z and I intend that to mean its API is not stable. There are going to be further breakages as I untangle the modules, unfortunately.

I'll make nice release notes about it, though.